src/routes/bulk/post.rs

```diff
@@ -87,18 +77,69 @@ impl Iterator for CreateBulkRequestBodyIterator {
 				item.set_from_email(email.clone());
 			}
 
-			if let Some(port) = self.body.smtp_port {
-				item.set_smtp_port(port);
-			}
+			item.set_smtp_port(self.body.smtp_ports[self.index]);
 
 			if let Some(proxy) = &self.body.proxy {
 				item.set_proxy(proxy.clone());
 			}
 
 			item.set_smtp_timeout(Duration::from_secs(SMTP_TIMEOUT));
 
-			self.index = bounded_index;
+			self.index += 1;
+			Some(item)
+		} else {
+			None
+		}
+	}
+}
+
+/// Endpoint request body.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+struct CreateBulkRequestBody {
+	input_type: String,
+	input: Vec<String>,
```

Comment:
Nothing to do in this PR, but I would like to do some forward-thinking. We already have CSV download of results, at some point we would need CSV upload of emails for the request too.

In that case, the `input` would be a string, referrering to the URL of the CSV. Do you have ideas how to structure this? 

(again no need to do anything now, but just want to make this change easy for future work).

Reply:
I think this field would probably become an enum with two variants namely `Vec<String> | String`. Then the enum would implement a trait method to return `Result<Vec<String>, Err>`. For the url case it would fetch the values from the file system.